### PR TITLE
test: move Modal to RTL

### PR DESCRIPTION
**Pull request by AhamedFrontEndDeveloper**
Tests for Modal component

part of https://github.com/react-bootstrap/react-bootstrap/issues/6156

### CODE REVIEW

**Comment by AhamedFrontEndDeveloper** [comment]
@kyletsang Can you please check this PR.

File: test/ModalSpec.tsx [hunk 1]
```diff
@@ -22,219 +21,228 @@ describe('<Modal>', () => {
...
-      .should.equal('Message');
+    
+    expect(getByTestId('modal').querySelector('strong').textContent).to.equal('Message')
   });
 
   it('Should sets `display: block` to `div.modal` when animation is false', () => {
-    const node = mount(
-      <Modal show animation={false}>
+    const ref = React.createRef<ModalProps>();
+    const { getByTestId } = render(
```
Comment (golota60) [inline test/ModalSpec.tsx:35]: I think this test is not using `getByTestId`

File: test/ModalSpec.tsx [hunk 2]
```diff
@@ -22,219 +21,228 @@ describe('<Modal>', () => {
...
       <Modal show onHide={doneOp}>
         <strong>Message</strong>
       </Modal>,
     )
-      .find('div.modal') // the modal-dialog element is pointer-events: none;
-      .simulate('click');
+
+    // the modal-dialog element is pointer-events: none;
+    fireEvent.click(getByRole('dialog'));
+    expect(onHideSpy).to.have.been.called;
```
Comment (golota60) [inline test/ModalSpec.tsx:57]: `onHideSpy` is not defined anywhere in the test

File: test/ModalSpec.tsx [hunk 3]
```diff
@@ -22,219 +21,228 @@ describe('<Modal>', () => {
...
     const noOp = () => {};
-    const wrapper = mount(
+    const { getByRole } = render(
       <Modal show onHide={noOp} backdrop="static" keyboard={false}>
         <strong>Message</strong>
       </Modal>,
     );
 
     const event = new KeyboardEvent('keydown', { keyCode: 27 });
     document.dispatchEvent(event);
```
Comment (golota60) [inline test/ModalSpec.tsx:94]: can this be replaced with `testing-library`'s `fireEvent.keyDown` function?

File: test/ModalSpec.tsx [hunk 4]
```diff
@@ -244,59 +252,71 @@ describe('<Modal>', () => {
...
     const increment = sinon.spy();
     let modal;
 
-    const instance = mount(
+    const { getByTestId, getByRole, rerender } = render(
       <Modal
         show
+        data-testid="modal"
         style={{ transition: 'opacity 1s linear' }}
+        onHide={() => {}}
```
Comment (golota60) [inline test/ModalSpec.tsx:303]: is defining `onHide` needed?

File: test/ModalSpec.tsx [hunk 5]
```diff
@@ -325,59 +345,57 @@ describe('<Modal>', () => {
...
       </Modal>,
     );
-    wrapper.assertSingle(
-      'div.modal.show[role="dialog"][aria-labelledby="modal-title"]',
-    );
+
+    expect(getByRole('dialog').getAttribute('aria-labelledby')).to.equal('modal-title');
   });
 
   it('Should call onEscapeKeyDown when keyboard is true', () => {
```
Comment (golota60) [inline test/ModalSpec.tsx:395]: can this test also use `fireEvent.keyDown` in line 405?

File: test/ModalSpec.tsx [hunk 6]
```diff
@@ -392,7 +410,7 @@ describe('<Modal>', () => {
   it('Should not call onEscapeKeyDown when keyboard is false', () => {
```
Comment (golota60) [inline test/ModalSpec.tsx:410]: can this test also use `fireEvent.keyDown` in line 425

File: test/ModalSpec.tsx [hunk 7]
```diff
@@ -1,18 +1,17 @@
...
+import { fireEvent, render, cleanup } from '@testing-library/react';
 import ModalManager from '@restart/ui/ModalManager';
-import Modal from '../src/Modal';
+import Modal, { ModalProps } from '../src/Modal';
 
 describe('<Modal>', () => {
   afterEach(() => {
-    // make sure the dangling portal elements get cleaned up
-    document.body.innerHTML = '';
+    cleanup();
```
Comment (golota60) [inline test/ModalSpec.tsx:8]: is this `afterEach` still needed? https://testing-library.com/docs/react-testing-library/api/#cleanup

**Review by golota60 — COMMENTED** [review]
looking good! left a few little comments here and there

File: test/ModalSpec.tsx (hunk 7 above)
Comment (AhamedFrontEndDeveloper) [reply test/ModalSpec.tsx:8]: As defined in the docs, It is recommended to use cleanup method so preferred it.
`Failing to call cleanup when you've called render could result in a memory leak and tests which are not "idempotent" (which can lead to difficult to debug errors in your tests).`
https://testing-library.com/docs/react-testing-library/api/#cleanup

**Review by golota60 — APPROVED** [review]
lgtm! 👍

File: test/ModalSpec.tsx (hunk 7 above)
Comment (kyletsang) [reply test/ModalSpec.tsx]: My understanding is that this should be done automatically.  Was this not working?

> Please note that this is done automatically if the testing framework you're using supports the afterEach global and it is injected to your testing environment (like mocha, Jest, and Jasmine). If not, you will need to do manual cleanups after each test.

Comment (AhamedFrontEndDeveloper) [reply test/ModalSpec.tsx]: I missed it. Will remove the after each hooks.

**Review by kyletsang — APPROVED** [review]

**Comment by kyletsang** [comment]
Thanks

**Pull request merged**